### ISSUE TITLE
assumeAlso fails when there are symbolic size matrices around

### ISSUE THREAD

**Issue by cbm755** [cbm755/octsympy#201]
````
syms n integer
B = sym('B', [n n])
syms x positive
x = assumeAlso(x, 'integer')  % gives error about MatrixSymbol
````

**Comment by cbm755** [comment]
Well its a bit better now b/c of changes to `x = assumeAlso`, but still happens with:
````
assumeAlso(x, 'integer')
````

**Comment by cbm755** [comment]
Fixed this for SymPy > 0.7.5.
